Design/purity-checker-proposal.md

```diff
@@ -0,0 +1,103 @@
+## !First Concept! for the Purity Checker
+
+We want a call of `checkValidity()` on an arbitrary `ExpEmbedding` node to return `true` if all `assert` statements present contain only pure expressions, and `false` as soon as one of them is impure. We split the logic into traversal and validation, currently only focussing on the validation of expressions appearing inside of `assert` nodes.
+
+Every `ExpEmbedding` implements a new `children` property and the functions `checkValidity` and `checkOwnValidity`.
```

Comment:
I wouldn't introduce something new; rather, we have a `subexpressions` property on many ExpEmbeddings already.